udf-examples/src/main/cpp/benchmarks/synchronization/synchronization.cpp

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (c) 2021, NVIDIA CORPORATION.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "synchronization.hpp"
+
+#include <cudf/utilities/error.hpp>
+
+#include <rmm/cuda_stream_view.hpp>
+#include <rmm/device_buffer.hpp>
+
+cuda_event_timer::cuda_event_timer(benchmark::State& state,
+                                   bool flush_l2_cache,
+                                   rmm::cuda_stream_view stream)
+  : stream(stream), p_state(&state)
+{
+  // flush all of L2$
+  if (flush_l2_cache) {
```

Comment:
Is this done to help the timer code be more consistent? My guess yes, but it wasn't entirely clear. Maybe expanding the comment?

Reply:
It's to make the benchmark more consistent by reducing the impact of caching from run-to-run.  Standard stuff.  This file and the benchmark fixture file were lifted from cudf essentially verbatim because I couldn't easily access them during the build otherwise.